pennylane/qnode.py

```diff
@@ -29,6 +29,24 @@
 from pennylane.tape import QuantumTape, make_qscript
 
 
+def _convert_to_interface(res, interface):
+    """
+    Recursively convert res to the given interface.
+    """
+    interface = INTERFACE_MAP[interface]
+
+    if interface in ["Numpy", "autograd"]:
+        return res
+
+    if isinstance(res, (list, tuple)):
+        return type(res)(_convert_to_interface(r, interface) for r in res)

```

Comment:
is this conversion necessary? I think `qml.math.asarray` will convert lists and tuples properly

Reply:
ah right, jumped the gun. sounds good!